tests/defer/tests.py

```diff
@@ -271,3 +272,21 @@ def test_custom_refresh_on_deferred_loading(self):
             # access of any of them.
             self.assertEqual(rf2.name, 'new foo')
             self.assertEqual(rf2.value, 'new bar')
+
+    def test_defaul_defer_fields(self):
+        """
+        The model `Publication` has two fields which are marked as deferred:
+        `title` and `user`. This two fields must be deferred during getting
+        object from DB
+        """
+        u = User.objects.create()
+        p = Publication()
+        p.user = u
+        p.save()
+        p = Publication.objects.get()
+        publication = Publication.objects.defer('text').get()
+        self.assertEqual(len(p.get_deferred_fields()), 2)
+        self.assertEqual(len(publication.get_deferred_fields()), 3)
+        self.assertEqual(p.user, u)
+        self.assertEqual(publication.text, "Test text")
+        self.assertEqual(p.title, "Test Title")
```

Comment:
`refresh_from_db` doesn't currently work with default deferred fields the same way it does with regular `.defer()` fields. Try adding this to the test to show the problem:

        # this works as expected, where publication was loaded with defer('text'):
        publication.text = "Refresh text"
        p.refresh_from_db()
        self.assertEqual(p.text, "Test text")
        
        # this fails, where p.title is deferred by default:
        p.title = "Refresh title"
        p.refresh_from_db()
        self.assertEqual(p.title, "Test Title")

The second part fails at `self.assertEqual(p.title, "Test Title")`:

    AssertionError: 'Refresh title' != 'Test Title'
    - Refresh title
    + Test Title

Reply:
Has `refresh_from_db` ever worked properly? From my personal experience we found the built in function buggy. We tend to just reload via `Object.model.get(id=old_object.id)` in our code.